src/routes/MapExplorer/MapExplorer.connector.js

```diff
@@ -103,28 +103,27 @@ export function mapStateToProps (state, props) {
     fetchPublicCommunitiesParam,
     filters: state.MapExplorer.clientFilterParams,
     members,
-    // TODO: show loading spinner while pending
     pending: state.pending[FETCH_POSTS_MAP],
     posts,
     publicCommunities,
     routeParams,
-    showDrawer,
+    hideDrawer,
     topics,
     zoom: centerLocation ? 10 : 0
   }
 }
 
 export function mapDispatchToProps (dispatch, props) {
   const routeParams = get('match.params', props)
-  const querystringParams = getQuerystringParam(['showDrawer', 't', 'network'], null, props)
+  const querystringParams = getQuerystringParam(['hideDrawer', 't', 'network'], null, props)
   return {
     fetchMembers: (params) => () => dispatch(fetchMembers({ ...params })),
     fetchPosts: (params) => () => dispatch(fetchPosts({ ...params })),
     fetchPublicCommunities: (params) => () => dispatch(fetchPublicCommunities({ ...params })),
     showDetails: (postId) => dispatch(push(postUrl(postId, { ...routeParams, view: 'map' }, querystringParams))),
     showCommunityDetails: (communityId) => dispatch(push(communityMapDetailUrl(communityId, { ...routeParams, view: 'map' }, querystringParams))),
     gotoMember: (memberId) => dispatch(push(personUrl(memberId, routeParams.slug, routeParams.networkSlug))),
-    toggleDrawer: (visible) => dispatch(push(addQuerystringToPath(baseUrl({ ...routeParams, view: 'map' }), { ...querystringParams, showDrawer: visible }))),
+    toggleDrawer: (hidden) => dispatch(push(addQuerystringToPath(baseUrl({ ...routeParams, view: 'map' }), { ...querystringParams, hideDrawer: hidden }))),
```

Comment:
I'm noticing that in the all and public map views, closing the drawer directs to `/map` rather than  `/all/map` or `/public/map`. I believe this can be solved by adding context to this function.

```
toggleDrawer: (hidden, context) => dispatch(push(addQuerystringToPath(baseUrl({ ...routeParams, context, view: 'map' }), { ...querystringParams, hideDrawer: hidden }))),
```

Reply:
huh, its working for me